### PR TITLE
Change distributor BucketDBUpdater to handle multiple bucket spaces

### PR DESCRIPTION
**Pull request by toregge**
@geirst, @vekterli: please review

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

File: storage/src/vespa/storage/distributor/bucketdbupdater.cpp [hunk 1]
```diff
@@ -451,27 +447,28 @@ BucketDBUpdater::addBucketInfoForNode(
...
 
 void
-BucketDBUpdater::findRelatedBucketsInDatabase(uint16_t node, const document::BucketId& bucketId,
+BucketDBUpdater::findRelatedBucketsInDatabase(uint16_t node, const document::Bucket& bucket,
                                               BucketListMerger::BucketList& existing)
 {
+    auto &distributorBucketSpace(_bucketSpaceComponent.getBucketSpaceRepo().get(bucket.getBucketSpace()));
     std::vector<BucketDatabase::Entry> entries;
-    _bucketSpaceComponent.getBucketDatabase().getAll(bucketId, entries);
+                                 distributorBucketSpace.getBucketDatabase().getAll(bucket.getBucketId(), entries);
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/bucketdbupdater.cpp]: Indentation here is a bit on the generous side?

**Review by vekterli — APPROVED** [review]
LGTM, see formatting nit

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**